src/controllers/csi/driver/server.go

```diff
@@ -181,6 +181,9 @@ func (svr *CSIDriverServer) NodeUnpublishVolume(ctx context.Context, req *csi.No
 		}
 	}
 	log.Info("VolumeID not present in the database", "volumeID", volumeInfo.VolumeID, "targetPath", volumeInfo.TargetPath)
+	if err != svr.mounter.Unmount(volumeInfo.TargetPath) {
```

Comment:
why this weird check?

Reply:
this only happens if there is no info on the mount in the database, so I try to unmount in hopes that it will just work and let the pod die/restart

Reply:
I noticed that I was dumb, fixed it 😅